tensorflow/python/kernel_tests/variable_scope_test.py

```diff
@@ -721,6 +721,48 @@ def device_func(op):
     self.assertEqual(varname_type[0], ("x", dtypes.float32))
     self.assertEqual(varname_type[1], ("y", dtypes.int64))
 
+  def testGetCollection(self):
+    with self.test_session():
+      a = variable_scope.get_variable("a", [])
+      b = variable_scope.get_variable("b", [], trainable=False)
+      with variable_scope.variable_scope("foo_") as scope:
+        a = variable_scope.get_variable("a", [])
+        b = variable_scope.get_variable("b", [], trainable=False)
+      with variable_scope.variable_scope("foo") as scope:
+        a = variable_scope.get_variable("a", [])
+        b = variable_scope.get_variable("b", [], trainable=False)
+        self.assertEqual(
+            [v.name for v in scope.get_collection(
+              ops.GraphKeys.TRAINABLE_VARIABLES)],
+            ["foo/a:0"])
+        self.assertEqual(
+            [v.name for v in scope.get_collection(
+              ops.GraphKeys.GLOBAL_VARIABLES)],
+            ["foo/a:0", "foo/b:0"])
+      scope = variable_scope.get_variable_scope()
+      self.assertEqual(
+          [v.name for v in scope.get_collection(
+            ops.GraphKeys.GLOBAL_VARIABLES)],
+          ["a:0", "b:0", "foo_/a:0", "foo_/b:0", "foo/a:0", "foo/b:0"])
+      self.assertEqual(
+          [v.name for v in scope.get_collection(
+            ops.GraphKeys.TRAINABLE_VARIABLES)],
+          ["a:0", "foo_/a:0", "foo/a:0"])
+
+  def testGetTrainableVariables(self):
+    with self.test_session():
+      a = variable_scope.get_variable("a", [])
+      b = variable_scope.get_variable("b", [], trainable=False)
+      scope = variable_scope.get_variable_scope()
+      self.assertEqual(scope.trainable_variables()[0].name, "a:0")
+
+  def testGetGlobalVariables(self):
+    with self.test_session():
+      a = variable_scope.get_variable("a", [])
+      b = variable_scope.get_variable("b", [], trainable=False)
```

Comment:
Again, please add a variable and then open a scope, and check that the first one isn't in the ones we get.